src/manage.c

```diff
@@ -610,4 +635,145 @@ u8 tracking_channels_ready()
   return n_ready;
 }
 
+/** Queue a request to start up tracking and decoding for the specified sid.
+ *
+ * \note This function is thread-safe and non-blocking.
+ *
+ * \param startup_params    Struct containing startup parameters.
+ *
+ * \return true if the request was successfully submitted, false otherwise.
+ */
+bool tracking_startup_request(const tracking_startup_params_t *startup_params)
+{
+  bool result = false;
+  if(chMtxTryLock(&tracking_startup_mutex))
+  {
+    result = tracking_startup_fifo_write(&tracking_startup_fifo,
+                                         startup_params);
+
+    Mutex *m = chMtxUnlock();
+    assert(m == &tracking_startup_mutex);
+  }
+
+  return result;
+}
+
+/** Read tracking startup requests from the FIFO and attempt to start
+ * tracking and decoding.
+ */
+static void manage_tracking_startup(void)
+{
+  tracking_startup_params_t startup_params;
+  while(tracking_startup_fifo_read(&tracking_startup_fifo, &startup_params)) {
+
+    acq_status_t *acq = &acq_status[sid_to_global_index(startup_params.sid)];
+
+    /* Make sure a tracking channel and a decoder channel are available */
+    u8 chan = manage_track_new_acq(startup_params.sid);
+    if (chan == MANAGE_NO_CHANNELS_FREE) {
+
+      /* No channels are free to accept our new satellite :( */
+      /* TODO: Perhaps we can try to warm start this one
+       * later using another fine acq.
+       */
+      if (startup_params.cn0_init > ACQ_RETRY_THRESHOLD) {
+        acq->score[ACQ_HINT_PREV_ACQ] =
+            SCORE_ACQ + (startup_params.cn0_init - ACQ_THRESHOLD);
+        acq->dopp_hint_low = startup_params.carrier_freq - ACQ_FULL_CF_STEP;
+        acq->dopp_hint_high = startup_params.carrier_freq + ACQ_FULL_CF_STEP;
+      }
+
+      continue;
+    }
+
+    /* Transition to tracking. */
+    u32 track_count = nap_timing_count() + 20000;
```

Comment:
Old code I just relocated. But it looks like the upper 32 bits are only maintained in software
```
/** Get the current NAP internal sample clock count.
 * NAP's internal count of sample clocks + (number of NAP's counter rollovers)
 * times 2^32. NAP's internal sample clock counter is 32 bits wide - at a
 * 16.368MHz sample clock frequency it rolls over approximately every 262
 * seconds.
 *
 * \return NAP's internal count of sample clocks +
 *               (total number of NAP counter rollovers) * 2^32.
 */
u64 nap_timing_count(void)
{
  static u32 rollover_count = 0;
  static u32 prev_count = 0;

  u8 temp[4] = { 0, 0, 0, 0 };

  nap_xfer_blocking(NAP_REG_TIMING_COUNT, 4, temp, temp);

  u32 count = (temp[0] << 24) | (temp[1] << 16) | (temp[2] << 8) | temp[3];

  if (count < prev_count)
    rollover_count++;

  prev_count = count;

  return (u64)count | ((u64)rollover_count << 32);
}
```